### PR TITLE
Fix - Skip excluded payment methods from activation toggling 

### LINKED ISSUE #2533: Adyen POS Coud API is always activated after a new installation
**Describe the bug**
"Adyen POS Coud API" does not keep the configuration and every time we make a new installation of the plugin it appears as activated even if we have previously disabled it.

**To Reproduce**
With the plugin already installed
1. Payment Methods --> Adyen --> Accepting Payments --> In-person payments
2. Change "Enabled" to No
3. Exec `composer install`
4. See In-person is enabled again and the method is showed on the frontend

**Magento version**
2.4.5-p6

**Plugin version**
9.2.0

### PR DESCRIPTION
**Pull request by hossam-adyen**
**Description**

After introducing  the recurring setup with this [PR](https://github.com/Adyen/adyen-magento2/pull/2476), if a customer set his POS disabled then ran `setup:upgrade`, the disabled POS payment method will be enabled and this is fixing that by filtering out POS payment

Screenshot 2024-02-12 at 3 52 05 PM

**Tested scenarios**

Fixes #2533

### CODE REVIEW

File: Helper/PaymentMethods.php [hunk 1]
```diff
@@ -153,6 +153,11 @@ public function togglePaymentMethodsActivation(?bool $isActive =null): array
         }
 
         foreach ($this->getAdyenPaymentMethods() as $paymentMethod) {
+            //skip POS from toggling as this config is visual via Admin panel
+            if ($paymentMethod == Config::XML_ADYEN_POS_CLOUD) {
```
Comment (candemiralp) [inline Helper/PaymentMethods.php:157]: This skip case should be extended with [EXCLUDED_PAYMENT_METHODS](https://github.com/Adyen/adyen-magento2/blob/5e5bdd63e00974c9ee2edc9f9bdfdd096e5eaba5/Model/Config/Backend/PaymentMethodsStatus.php#L35) const array. I think the usage of this array was skipped on [this](https://github.com/Adyen/adyen-magento2/pull/2476) PR.

Comment (hossam-adyen) [reply Helper/PaymentMethods.php:157]:
```suggestion
            if (in_array($paymentMethod, PaymentMethodsStatus::EXCLUDED_PAYMENT_METHODS]) {
```

File: Test/Unit/Helper/PaymentMethodsTest.php [hunk 2]
```diff
@@ -170,7 +170,8 @@ public function testTogglePaymentMethodsActivation()
                 [
                     'adyen_cc' => [],
                     'adyen_oneclick' => [],
-                    'adyen_cc_vault' => []
+                    'adyen_cc_vault' => [],
+                    'adyen_pos_cloud' => [],
```
Comment (hossam-adyen) [inline Test/Unit/Helper/PaymentMethodsTest.php:174]:
```suggestion
                    'adyen_pos_cloud' => [],
                    'adyen_moto' => [],
                    'adyen_pay_by_link' => [],
```

File: Helper/PaymentMethods.php [hunk 3]
```diff
@@ -153,6 +154,10 @@ public function togglePaymentMethodsActivation(?bool $isActive =null): array
         }
 
         foreach ($this->getAdyenPaymentMethods() as $paymentMethod) {
+            if (in_array($paymentMethod, PaymentMethodsStatus::EXCLUDED_PAYMENT_METHODS) {
```
Comment (hossam-adyen) [inline Helper/PaymentMethods.php:157]:
```suggestion
            if (in_array($paymentMethod, PaymentMethodsStatus::EXCLUDED_PAYMENT_METHODS)) {
```

File: Model/Config/Backend/PaymentMethodsStatus.php [hunk 4]
```diff
@@ -27,50 +23,25 @@
...
-            if (in_array($adyenPaymentMethod, self::EXCLUDED_PAYMENT_METHODS)) {
-                continue;
-            }
-
-            $this->configWriter->save(
-                'payment/' . $adyenPaymentMethod . '/active',
-                $value,
-            );
-        }
+        $this->paymentMethodsHelper->togglePaymentMethodsActivation($this->getValue());
```
Comment (hossam-adyen) [inline Model/Config/Backend/PaymentMethodsStatus.php:44]: `togglePaymentMethodsActivation()` expects a boolean and I think `$this->getValue()` is a string with value either `'1'` or `'0'`

**Review by candemiralp — APPROVED** [review]

**Review by candemiralp — APPROVED** [review]

**Review by khushboo-singhvi — APPROVED** [review]

**Pull request merged**